### ISSUE TITLE
Add file argument to get_nml_value

### ISSUE THREAD

**Issue by jsta** [USGS-R/glmtools#233]
I am thinking that it would be useful to add a `file` argument to `get_nml_value` similar to the other `get_*` functions.

**Comment by lawinslow** [comment]
So it would operate directly on a file instead of a loaded nml object?

**Comment by jsta** [comment]
I think either-or. Consider this block from `get_hypsography`

```
  # if both are passed, glm_nml is used and 'file' is ignored
  if (missing(file) & missing(glm_nml)){stop('glm.nml file path OR glm_nml must be specified')}
  
  if (missing(glm_nml)){
    glm_nml <- read_nml(file)
  }
```

**Comment by lawinslow** [comment]
Would it then also overwrite the old original file as well? Basically, 

````
  # if both are passed, glm_nml is used and 'file' is ignored
  if (missing(file) & missing(glm_nml)){stop('glm.nml file path OR glm_nml must be specified')}
  
  if (missing(glm_nml)){
    glm_nml <- read_nml(file)
  }


  #do a bunch of stuff to glm_nml

  write_nml(glm_nml, file)
````

**Comment by jsta** [comment]
I don't follow. Why would `get_nml_value` be doing things to `glm_nml`? My understanding was that it simply retrieved values for inspection unchanged. Are you thinking of a function that would combine an nml file and and local nml object and write a new nml file (aka "`combine_nml`")?

**Comment by lawinslow** [comment]
Ah yeah, ignore that. For some reason I was thinking about a function that muted the file in place. Like we were talking about `set_nml`. 

That being said, wouldn't it make sense to implement the parallel version in `set_nml`? A file could be specified and it would overwrite the NML with a new one with the new variable set. This would eliminate alot of code that reads, alters, then writes NML files.

**Comment by jsta** [comment]
I am about to submit a PR with this ability added. I was thinking only of reading values not of writing or setting.

I think what you are talking about is more along the lines of #231?

**Comment by lawinslow** [comment]
My vote is to parallel this functionality in the `set_nml` function. But I'll leave this up to @jread-usgs if he wants to merge the PR as-is.

**Comment by jread-usgs** [comment]
Awesome work @jsta 
I agree w/ Luke that it would be best to parallel this, but in a different PR if possible. I haven't been able to review (in all-day meetings), but will soon.

**Comment by jread-usgs** [comment]
Yes, I like this idea, and like the parallel idea @lawinslow proposed for `set_nml`
